FWCore/Framework/src/EventProcessor.cc

```diff
@@ -1747,7 +1788,10 @@ namespace edm {
   }
 
   void EventProcessor::setExceptionMessageLumis(std::string& message) {
-    exceptionMessageLumis_ = message;
+    bool expected = false;
+    if (exceptionMessageLumisIsSet_.compare_exchange_strong(expected,true)) {
```

Comment:
This is not technically thread safe. The `exceptionMessageLumisIsSet_` is not functioning as a proper guard since it does not guarantee that after writing to `exceptionMessageLumis_` that memory is synchronized across all threads that might read it. In practice, when the `tbb::task` that called `setExceptionMessageLumis` completes, it will synchronize all memory written during the task's exectution.

The better way to do this is to mae `exceptionMessageLumis_ ` a `std::atomic<std::string*>` and handle memory cleanup at the point which looks at `exceptionMessageLumis_`.

Reply:
Agreed. Although nothing was actually reading it in the concurrent parts of the code, but I agree it is a landmine we might possibly hit later. I thought of a simpler fix. The string is always the same now.  It is simpler just to delete the string data member entirely. It is not needed. All we need is an atomic<bool> data member. Fixed in the commit I just pushed.